script.js

```diff
@@ -1,4 +1,153 @@
+
+var inputName = function (name) {
+  return `Hello ${name}, please input 'scissors' 'paper' or 'stone' to start playing the game.`;
+};
+
+var getIcon = function (hand) {
+  if (hand == "scissors" || hand == "reversed scissors") {
+    return "✂️";
+  }
+  if (hand == "paper" || hand == "reversed paper") {
+    return "🗒";
+  }
+  if (hand == "stone" || hand == "reversed stone") {
+    return "🪨";
+  }
+};
+
+// normal game mode
+var playNormalGame = function (playerHand, computerHand) {
+  if (
+    (playerHand == "scissors" && computerHand == "paper") ||
+    (playerHand == "paper" && computerHand == "stone") ||
+    (playerHand == "stone" && computerHand == "scissors")
+  ) {
+    return "win";
+  }
+  if (
+    (playerHand == "scissors" && computerHand == "stone") ||
+    (playerHand == "paper" && computerHand == "scissors") ||
+    (playerHand == "stone" && computerHand == "paper")
+  ) {
+    return "lose";
+  }
+  if (
+    (playerHand == "scissors" && computerHand == "scissors") ||
+    (playerHand == "paper" && computerHand == "paper") ||
+    (playerHand == "stone" && computerHand == "stone")
+  ) {
+    return "draw";
+  } else {
+    return "error";
+  }
+};
+
+// reversed game mode
+var playReversedGame = function (playerHand, computerHand) {
+  if (
+    (playerHand == "scissors" && computerHand == "stone") ||
+    (playerHand == "paper" && computerHand == "scissors") ||
+    (playerHand == "stone" && computerHand == "paper")
+  ) {
+    return "win";
+  }
+  if (
+    (playerHand == "scissors" && computerHand == "paper") ||
+    (playerHand == "paper" && computerHand == "stone") ||
+    (playerHand == "stone" && computerHand == "scissors")
+  ) {
+    return "lose";
+  }
+  if (
+    (playerHand == "scissors" && computerHand == "scissors") ||
+    (playerHand == "paper" && computerHand == "paper") ||
+    (playerHand == "stone" && computerHand == "stone")
+  ) {
+    return "draw";
+  } else {
+    return "error";
+  }
+};
+
+// definiing round outcomes count
+var roundCount = -1;
+var winCount = 0;
+var loseCount = 0;
+var drawCount = 0;
+var prevWinner = 0;
+
+var reverseGameMode = false;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (roundCount == -1) {
+    roundCount += 1;
+    var playerName = input;
+    return inputName(playerName);
+  } else {
+    // computer's choice
+    var computer = computerChoice();
+    var computerIcon = getIcon(computer);
+    var playerIcon = getIcon(input);
+
+    // start of game
+    roundCount += 1;
+
+    if (input == "reverse") {
+      reverseGameMode = true;
+      return "Start of reverse game mode";
+    }
+
+    if (input == "normal") {
+      reverseGameMode = false;
+      return "Start of normal game mode";
+    }
+
+    if (reverseGameMode) {
+      var outcome = playReversedGame(input, computer);
+    } else {
+      outcome = playNormalGame(input, computer);
+    }
+
+    if (outcome == "win") {
+      winCount += 1;
+      prevWinner = "you";
```

Comment:
good job on implementing the korean sps version as well! It's not necessarily difficult but requires time to think through and execute it.